base/shell/explorer/traydeskbtn.cpp

```diff
@@ -0,0 +1,438 @@
+/*
+ * PROJECT:     ReactOS Explorer
+ * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
+ * PURPOSE:     Show Desktop tray button implementation
+ * COPYRIGHT:   Copyright 2006-2007 Thomas Weidenmueller <w3seek@reactos.org>
+ *              Copyright 2018-2022 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ *              Copyright 2023 Ethan Rodensky <splitwirez@gmail.com>
+ */
+
+#include "traydeskbtn.h"
+#include <limits.h>
+
+#define IDI_SHELL32_DESKTOP 35
+#define IDI_IMAGERES_DESKTOP 110
+
+#define SHOW_DESKTOP_TIMER_ID 999
+#define SHOW_DESKTOP_TIMER_INTERVAL 200
+
+CTrayShowDesktopButton::CTrayShowDesktopButton() :
+    m_nClickedTime(0),
+    m_inset({2, 2}),
+    m_icon(NULL),
+    m_highContrastMode(FALSE),
+    m_drawWithDedicatedBackground(FALSE),
+    m_bHovering(FALSE),
+    m_bPressed(FALSE),
+    IsHorizontal(FALSE)
+{
+}
+
+BOOL
+CTrayShowDesktopButton::GetTaskbar(OUT HWND* taskbarWnd)
+{
+    *taskbarWnd = NULL;
+    auto parent = GetParent();
+    if (!parent)
+        return FALSE;
+
+    parent = parent.GetParent();
```

Comment:
This parent = parent. .... is for getting the "grand-parent" of the desktop button (which, we hope, should be the taskbar window)?

Reply:
Yes, that's correct.

Reply:
Should we make sure this is always correct?

`ASSERT((HWND)parent == FindWindowExW(NULL, NULL, "Shell_TrayWnd", NULL));`